### PR TITLE
Hardening for 6.8

### PR DESCRIPTION
**Pull request by damien-urruty-sonarsource**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/vcs/DefaultVcsService.kt [hunk 1]
```diff
@@ -58,14 +58,20 @@ class DefaultVcsService @NonInjectable constructor(private val project: Project,
...
         }
         if (repositories.size > 1) {
             logger.warn("Several candidate Vcs repositories detected for module $module, cannot choose one")
             return null
         }
         val repo = repositories.first()
-        return repo.electBestMatchingServerBranchForCurrentHead(validConnectedEngine.getServerBranches(projectKey))
+        val resolvedBranch =
+            repo.electBestMatchingServerBranchForCurrentHead(validConnectedEngine.getServerBranches(projectKey))
+        if (resolvedBranch == null) {
```
Comment (henryju) [inline src/main/java/org/sonarlint/intellij/vcs/DefaultVcsService.kt:71]: I guess this change is not strictly related to the change of IJ api

**Review by henryju — APPROVED** [review]

File: src/main/java/org/sonarlint/intellij/vcs/DefaultVcsService.kt (hunk 1 above)
Comment (damien-urruty-sonarsource) [reply src/main/java/org/sonarlint/intellij/vcs/DefaultVcsService.kt:71]: No it was added for debug purposes when a test was failing. I will remove it
